src/ripple/net/HTTPStream.h

```diff
@@ -0,0 +1,148 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_NET_HTTPSTREAM_H_INCLUDED
+#define RIPPLE_NET_HTTPSTREAM_H_INCLUDED
+
+#include <ripple/core/Config.h>
+#include <ripple/net/HTTPClientSSLContext.h>
+
+#include <boost/asio/spawn.hpp>
+#include <boost/asio/strand.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+
+namespace ripple {
+
+class HTTPStream
+{
+public:
+    using request = boost::beast::http::request<boost::beast::http::empty_body>;
+    using parser = boost::beast::http::basic_parser<false>;
+
+    virtual ~HTTPStream() = default;
+
+    virtual boost::asio::ip::tcp::socket&
+    getStream() = 0;
+
+    virtual bool
+    connect(
```

Comment:
Consider decorating `connect` with the `[[nodiscard]]` attribute.

Reply:
Done